### ISSUE TITLE
Feature Request: Plugin System Extended Properties SubMenuItems entry

### ISSUE THREAD

**Issue by Andreas-Schniertshauer** [WinMerge/winmerge#851]
Is it possible to get a SubMenuItems extended properties entry that shows up when the Compare As\MenuCaption in the context menu is selected (SidePopup)? I ask because I want to pass the selected sub menu entry to my unpacker plugin so I don't need to open the Plugins\Plugin Settings dialog every time and change the parameter there. The selected entry may could be passed as plugin argument to the plugin e.g. as SubMenuItem=ID / or text.

**Comment by sdottaka** [comment]
I may not understand your request exactly, but what you see in the menu item is the filename of the plugin, or the value of the MenuCaption key in PluginExtendedProperies if version 2.16.13 or higher.

That is, the plugin itself should know its menu caption.

The IgnoreColumns.dll plugin behaves to ignore differences in columns 1-5 when renamed to IgnoreColumns_1-5.dll.
In this way, if you specify something like an argument in the file name of the plug-in and get the DLL name of the plug-in, you can get the equivalent of the argument.

Please refer to the source code of IgnoreColumns plugin. (GetDllFilename function)

**Comment by Andreas-Schniertshauer** [comment]
> I may not understand your request exactly, but what you see in the menu item is the filename of the plugin, or the value of the MenuCaption key in PluginExtendedProperies if version 2.16.13 or higher.

Yes, I understand, I used the MenuCaption to show a user friendly name for the plugin instead of the plugin dll name, e.g.
DLL: IgnoreColumns.dll
MenuCaption: Ignore the selected columns
SubMenuItems: 1-3,4;1,5;7-9,11
So with the SubMenuItems the menu would be

Plugins
  - Prediffer
    - Ignore the selected columns
      - 1-3,4
      - 1,5
      - 7-9,11

For my plugin it would be:
MenuCaption: XSLT Transformation
SubMenuItems: a.xsl;b.xsl;c.xsl

Plugins
  - Unpacker
    - XSLT Transformation
      - a.xsl
      - b.xsl
      - c.xsl
And the same in the context menu:
Compare As
  - XSLT Transformation
      - a.xsl
      - b.xsl
      - c.xsl

**Comment by sdottaka** [comment]
sorry. I misread it.

I think it's a good feature, but I don't think it will be implemented soon. Pull requests are welcome.

By the way, I would like to create a feature that is a bit like your request, a user-definable plugin + argument alias feature.
(It is possible to implement both as it does not conflict with your request)

For example, make it possible to define as follows

Alias name (menu item name): Ignore Columns from 1 to 5
Plugin Pipeline: IgnoreColumns 1-5

When the user selects the "Ignore Columns from 1 to 5" menu item, the IgnoreColumns plugin will be called and will be passed 1-5 as an argument.

Again, I don't think I'll implement it right away, so Pull Requests are welcome.

**Comment by Andreas-Schniertshauer** [comment]
May be a Alias name / Plugin arguments would easily be doable, I'll have a deeper look at the WinMerge source code.
